lib/clients/rtm/client.js

```diff
@@ -403,13 +415,15 @@ RTMClient.prototype._handleWsMessageViaEventHandler = function _handleWsMessageV
   messageType, message) {
   var replyTo;
 
+  if (!isUndefined(this.dataStore)) {
+    this.dataStore.handleRtmMessage(this.activeUserId, this.activeTeamId, messageType, message);
+  }
+
```

Comment:
So the api docs were right, though I find that a bad thing in this case. I feel like that'd be a cause of a high number of issues for anyone trying to make their own client to use with the RTM api (most people would probably throw an error if they got a message without a type that wasn't an error). That, or I'm crazy.

I have math to do so I won't be able to test this for a solid hour, but I'll be sure to throw my best worst-cases at it.